shell/platform/windows/flutter_windows_engine.cc

```diff
@@ -830,4 +984,56 @@ void FlutterWindowsEngine::OnChannelUpdate(std::string name, bool listening) {
   }
 }
 
+// Platform view methods:
+void FlutterWindowsEngine::RegisterPlatformViewType(
+    const std::string& type,
+    PlatformViewCreationContext factory) {
+  platform_view_types_.insert({type, factory});
+}
+
+void FlutterWindowsEngine::CreatePlatformView(PlatformViewCreationParams args) {
+  FML_LOG(ERROR) << "Creating a platform view of type \"" << args.view_type
+                 << "\"";
+  const auto itr = platform_view_types_.find(std::string(args.view_type));
+  if (itr == platform_view_types_.end()) {
+    FML_LOG(ERROR) << "View type not registered";
+    return;
+  }
+  args.user_data = itr->second.user_data;
+  pending_platform_views_.insert({args.id, {itr->second.factory, args}});
```

Comment:
The pending platform view is queued in a map rather than being immediately instantiated because an attempt to create the platform view before the Flutter window has begun to render may fail. The actual instantiation is performed by the present layers callback if there are any pending views.